scalableminds/webknossos
# unify admin task actions for task list and user annotation list (#1242)

This PR unifies the admin task actions for the task list view and a users annotation list view.
See discussion in https://discuss.webknossos.org/t/reopen-transfer-in-one-button/64 and #1242

I would've preferred to reuse some kind of view to accomplish this, but my Backbone-Fu was not sufficient to do this in a reasonable amount of time.
I've also noticed that for a users task list, there are no actions possible for an admin if the task is finished. I'm not sure whether this is needed and so I didn't add it yet as it would incur additional complexity (as the admin user task list on top of that is reused for the non-admin user task list view on the dashboard). I hope you understand what I mean, otherwise let's clarify in person!

Oh and btw I noticed that on removal a modal triggers a `modal:destroyed` event through `app.vent`, but all occurrences in the code listened to `modal:destroy`, so this functionality never seems to have worked. I fixed it and I hope this doesn't lead to errors anywhere due to functionality being invoked that wasn't invoked before ^^

### Mailable description of changes (needs to be understandable by webknossos mailing list people):
- Unify admin task actions for the task list view and a users annotation list view.

### Steps to test:
- Open the user list and click on "Show tracings" for one user that preferably has some open tasks
- In another tab open the task view and search for one of the task ids of the selected user - once the task is displayed toggle its details to see a list of all the corresponding annotations (and their actions in the actions drop-down)
- The possible actions for admins on one annotation should now be the same in the two tabs

- Also please test whether all the actions still work

### Issues:
- fixes #1242

------
